lib/spam_email/check.rb

```diff
@@ -1,11 +1,22 @@
 require 'mail'
+require 'public_suffix'
```

Comment:
I wonder if we can avoid adding another dependency.

Reply:
The public suffix list is the recommended way to retrieve the main domain. Think about domains like example.com and example.co.uk. Regular expressions and maintaining a list of all possible TLDs is too much effort.